### DISCUSSION TITLE
Wrong result for "which python" in Conda activated virtual env

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [jdx/mise#1407] [Q&A] [answered]
I'v installed Anaconda by downloading the official installer on my intel MacOs. All appears ok except when I activate a conda created virtual environment and then try `which python` I get the Mise installed python instead of the python from the virtual environment. 

The `path` indicates that the `which python` command should first find the python from the conda virtual environment since it's at the top of the path but that isn't the case.

```zsh
❯ path
/Users/user/anaconda3/envs/myenv/bin
/Users/user/.bin
/Users/user/.local/bin
/Users/user/.local/share/mise/installs/python/latest/bin
/Users/user/.local/share/mise/installs/go/1.21.5/go/bin
/Users/user/.local/share/mise/installs/go/1.21.5/packages/bin
...
/Users/user/.local/share/mise/installs/node/lts/bin
/Users/user/.local/share/mise/installs/bun/latest/bin
/Users/user/.local/share/mise/installs/neovim/nightly/bin
/Users/user/.local/share/pnpm
/Users/user/anaconda3/condabin
/Users/user/.opam/default/bin
/Users/user/.cabal/bin
/Users/user/.ghcup/bin
/Users/user/.cargo/bin
/Users/user/.juliaup/bin
/Users/user/go/bin
```
```zsh
❯ which python
/Users/user/.local/share/mise/installs/python/latest/bin/python
~/workspace/learning/python/test_conda                                               myenv
```

**Comment by jdx** [comment]
is this still the case on the latest version?

**Reply by Pilgrim1379** [reply]
Unfortunately Yes it still is. I'm on `mise 2024.1.14 macos-x64 (2024-01-09)`

**Comment by Marukome0743** [comment, marked as answer]
This appears to have been fixed by commit https://github.com/jdx/mise/commit/71ee6716c4516d69e5563115a2ca2391ba4bd16a, which changed activation so that PATH modifications made after `mise activate` take precedence over mise-managed tool paths. The change was first included in mise 2024.1.17.

I verified the original scenario on macOS arm64 with zsh 5.9, the current mise main branch, and an actual Miniconda environment. After activating both mise and a named conda environment, `which python`, `whence python`, zsh’s command cache, and `sys.prefix` all correctly pointed to the conda environment—even after the next mise prompt hook ran.

The original report used mise 2024.1.14, before this PATH behavior changed. I could also reproduce the old behavior on the current version by explicitly enabling `activate_aggressive`: conda’s Python was selected immediately after `conda activate`, but the next mise prompt hook moved the mise Python back to the front of PATH.

If this still occurs on a current version, please check:

```sh
mise settings get activate_aggressive
```

If it returns `true`, disabling that setting should allow the activated conda environment to remain ahead of mise in PATH.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
